job/header.sh

```diff
@@ -21,8 +21,26 @@ before_exit() {
   fi
 
   if [ "$is_success" == true ]; then
+    # "on_success" is only defined for the last task, so execute "always" only
+    # if this is the last task.
+    if [ "$(type -t on_success)" == "function" ]; then
```

Comment:
I've kept this handling the same as the current runSh. If there are any failures, it is ignored.